cmd/ctr/commands/containers/containers.go

```diff
@@ -194,16 +196,31 @@ func deleteContainer(ctx context.Context, client *containerd.Client, id string,
 	}
 	status, err := task.Status(ctx)
 	if err != nil {
+		if isNotFoundError(err) {
+			return container.Delete(ctx, opts...)
+		}
 		return err
 	}
 	if status.Status == containerd.Stopped || status.Status == containerd.Created {
 		if _, err := task.Delete(ctx); err != nil {
+			if isNotFoundError(err) {
+				return container.Delete(ctx, opts...)
+			}
 			return err
 		}
 		return container.Delete(ctx, opts...)
 	}
 	return fmt.Errorf("cannot delete a non stopped container: %v", status)
+}
 
+func isNotFoundError(err error) bool {
```

Comment:
I think we can just use errdefs.IsNotFound here because the client has unwrapped the error from grpc.

https://github.com/containerd/containerd/blob/9fffa8c71253d50a001cea5613bdf55d7273a3d6/process.go#L223

Reply:
Thanks, replaced with `errdefs.IsNotFound`